dev-tools/mage/pkgtypes.go

```diff
@@ -742,6 +743,18 @@ func runFPM(spec PackageSpec, packageType PackageType) error {
 		return err
 	}
 
+	packageName := spec.ServiceName
```

Comment:
Since there are no automated tests for the packaging, I suggest to add a comment here explaining why a dedicated package name and adding the conflicts is important. It might go a long way for the next developer touching this code to understand why this logic is important and what it solves.